pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -1888,19 +1869,27 @@ func (cc *XController) worker() {
 		//if everything passes then CanRun is set to true and AW is ready for dispatch
 		if !queuejob.Status.CanRun && (queuejob.Status.State != arbv1.AppWrapperStateActive) {
 			cc.ScheduleNext(queuejob)
-			return nil
+			//When an AW passes ScheduleNext gate then we want to progress AW to Running to begin with
+			//Sync queuejob will not unwrap an AW to spawn genericItems
+			if queuejob.Status.CanRun {
```

Comment:
To enter this `if` statement (line 1870), `queuejob.Status.CanRun` HAS to be `false` then this `if` will never be used. In the previous version this part was outside of this `if`, so why wast it put inside?

Reply:
When a new AW is entered in the system CanRun will always be false. previous versions had two round trips to the etcd to spawn items inside genericItems. This new change combines the two which increased the dispatch performance of MCAD.

Reply:
Understood